spec/system/signup/signup_spec.rb

```diff
@@ -0,0 +1,49 @@
+require_relative "../../../lib/google_mail"
+require_relative "../../../lib/eapol_test"
+
+feature "Signup" do
+  include_context "signup"
+
+  it "signs up successfully" do
+    gmail = GoogleMail.new
+
+    test_email = gmail.account_email.gsub(/@/, "+#{Time.now.to_i}@")
+
+    gmail.send(
+      "signup@wifi.service.gov.uk",
+      test_email,
+      "",
+      "",
+    )
+
+    body = nil
+
+    Timeout.timeout(20, nil, "Waited too long for signup email") do
+      loop do
```

Comment:
can we use a ```while``` loop or similar?

Reply:
Without `break`ing when we set `body` we will have an unnecessary delay with `while` or `until` (either before or after the final loop). If we are `break`ing, then the additional check at the top would be ignored, as we'll already have broken out. 

What do you think?

Reply:
That makes sense.